plugins/pkg/file/fs_test.go

```diff
@@ -16,28 +16,46 @@ package file
 
 import (
 	"os"
+	"sync"
 	"testing"
-	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
-func TestFileMtimeDetection(t *testing.T) {
-	defaultFs = newFS(2000 * time.Millisecond)
+func TestFileIsChanged(t *testing.T) {
+	changed := false
+	wg := sync.WaitGroup{}
+	once := sync.Once{}
 
-	tmpfile, _ := os.CreateTemp("", "example")
-	defer os.Remove(tmpfile.Name()) // clean up
+	watcher, err := NewWatcher()
 
-	f, err := Stat(tmpfile.Name())
 	assert.Nil(t, err)
-	assert.False(t, IsChanged(f))
-	time.Sleep(1000 * time.Millisecond)
+
+	tmpfile, _ := os.CreateTemp("./", "example")
```

Comment:
```suggestion
	tmpfile, _ := os.CreateTemp("", "example")
```
would be better so that the test case won't pollute the git repo.